### PR TITLE
🧹 refactor: remove organization id from request url

### PR DESCRIPTION
**Pull request by cosmos-explorer**
Currently the organizationId is used in the URLs when calling the API, but this is not needed. The current PR is to remove the organizationId in or API urls

### CODE REVIEW

File: modules/back-end/src/Api/Authentication/OpenApiConstants.cs [hunk 1]
```diff
@@ -3,4 +3,5 @@ namespace Api.Authentication;
 public static class OpenApiConstants
 {
     public const string PermissionStoreKey = "pemissions";
+    public const string OrgIdHeaderkey = "Organization";
```
Comment (deleteLater) [inline modules/back-end/src/Api/Authentication/OpenApiConstants.cs]: typo

image

File: modules/back-end/src/Api/Authentication/OpenApiHandler.cs [hunk 2]
```diff
@@ -58,6 +60,7 @@ protected override async Task<AuthenticateResult> HandleAuthenticateAsync()
 
                 var statements = policies.SelectMany(x => x.Statements);
                 Context.Items[OpenApiConstants.PermissionStoreKey] = statements;
+
```
Comment (deleteLater) [inline modules/back-end/src/Api/Authentication/OpenApiHandler.cs]: meaningless change

File: modules/back-end/src/Api/Controllers/ApiControllerBase.cs [hunk 3]
```diff
@@ -21,6 +22,12 @@ protected ICurrentUser CurrentUser
         get { return _currentUser ??= HttpContext.RequestServices.GetRequiredService<ICurrentUser>(); }
     }
 
+    private Guid? _orgId;
+    protected Guid OrgId
+    {
+        get { return _orgId ??= Guid.Parse(HttpContext.Request.Headers[OpenApiConstants.OrgIdHeaderkey]); }
```
Comment (deleteLater) [inline modules/back-end/src/Api/Controllers/ApiControllerBase.cs]: Use Guid.TryParse

File: modules/back-end/src/Api/Controllers/ProjectController.cs [hunk 4]
```diff
@@ -20,31 +20,29 @@ public async Task<ApiResponse<ProjectWithEnvs>> GetAsync(Guid projectId)
...
+        request.OrganizationId = OrgId;
 
         var projectWithEnvs = await Mediator.Send(request);
         return Ok(projectWithEnvs);
     }
 
-    [HttpPut("{id:guid}")]
-    public async Task<ApiResponse<ProjectVm>> UpdateAsync(Guid id, UpdateProject request)
+    [HttpPut]
+    public async Task<ApiResponse<ProjectVm>> UpdateAsync(UpdateProject request)
```
Comment (deleteLater) [inline modules/back-end/src/Api/Controllers/ProjectController.cs]: why this change

File: modules/front-end/src/app/core/components/project-drawer/project-drawer.component.ts [hunk 5]
```diff
@@ -105,7 +104,7 @@ export class ProjectDrawerComponent implements OnInit {
 
     if (this.isEditing) {
       this.projectService
-        .putUpdateProject(this.currentOrganizationId, {name, id: this.project.id})
+        .putUpdateProject({name, id: this.project.id})
```
Comment (deleteLater) [inline modules/front-end/src/app/core/components/project-drawer/project-drawer.component.ts]: - pass project id in query string
- rename `putUpdateProject` to `update`

File: modules/front-end/src/app/core/components/project-drawer/project-drawer.component.ts [hunk 6]
```diff
@@ -118,7 +117,7 @@ export class ProjectDrawerComponent implements OnInit {
         );
     } else {
       this.projectService
-        .postCreateProject(this.currentOrganizationId, {name})
+        .postCreateProject({name})
```
Comment (deleteLater) [inline modules/front-end/src/app/core/components/project-drawer/project-drawer.component.ts]: rename `postCreateProject` to `create`

File: modules/front-end/src/app/core/services/organization.service.ts [hunk 7]
```diff
@@ -29,17 +29,17 @@ export class OrganizationService {
   }
 
   updateOrganization(params: any): Observable<any> {
-    const url = `${this.baseUrl}/${params.id}`;
+    const url = `${this.baseUrl}`;
     return this.http.put(url, params);
   }
```
Comment (deleteLater) [inline modules/front-end/src/app/core/services/organization.service.ts]: - rename `updateOrganization` to `update`
- meaningless `url` variable, use `this.http.put(this.baseUrl, params)`

File: modules/front-end/src/app/features/safe/organizations/organization/organization.component.ts [hunk 8]
```diff
@@ -85,19 +85,19 @@ export class OrganizationComponent implements OnInit {
     const {id, initialized} = this.currentOrganization;
 
     this.isLoading = true;
-    this.organizationService.updateOrganization({name, id})
+    this.organizationService.updateOrganization({ name })
```
Comment (deleteLater) [inline modules/front-end/src/app/features/safe/organizations/organization/organization.component.ts]: rename `updateOrganization` to `update`

File: modules/front-end/src/app/features/safe/organizations/project/project.component.ts [hunk 9]
```diff
@@ -120,7 +118,7 @@ export class ProjectComponent implements OnInit {
       return;
     }
 
-    this.projectService.removeProject(this.currentOrganizationId, project.id).subscribe(() => {
+    this.projectService.removeProject(project.id).subscribe(() => {
```
Comment (deleteLater) [inline modules/front-end/src/app/features/safe/organizations/project/project.component.ts]: rename `removeProject` to `delete`

File: modules/back-end/src/Api/Controllers/ProjectController.cs (hunk 4 above)
Comment (cosmos-explorer) [reply modules/back-end/src/Api/Controllers/ProjectController.cs]: id is already in UpdateProject

File: modules/front-end/src/app/core/components/project-drawer/project-drawer.component.ts (hunk 5 above)
Comment (cosmos-explorer) [reply modules/front-end/src/app/core/components/project-drawer/project-drawer.component.ts]: why?
`pass project id in query string`

**Review by deleteLater — APPROVED** [review]
LGTM
